dhall-csv/src/Dhall/Csv.hs

```diff
@@ -25,42 +104,120 @@ import qualified Dhall.Import
+            \because it has type:                                                            \n\
+            \                                                                                \n\
+            \" <> insert (fromRight e (TypeCheck.typeOf e))
 
-instance Exception CompileError
+    displayException (NotAList e) =
+        Data.Text.unpack $
+            _ERROR <> ": Top level object must be of type ❰List❱                             \n\
+            \                                                                                \n\
+            \Explanation: To translate to CSV you must provide a list of records.            \n\
+            \Other types can not be translated directly.                                     \n\
+            \                                                                                \n\
+            \Expected an expression of type List {...} but instead got the following         \n\
+            \expression:                                                                     \n\
+            \                                                                                \n\
+            \" <> insert e <>                                                               "\n\
+            \which has type:                                                                 \n\
+            \                                                                                \n\
+            \" <> insert (fromRight e (TypeCheck.typeOf e))
+
+    displayException (NotARecord e) =
+        Data.Text.unpack $
+            _ERROR <> ": Elements of the top-level list must be records                      \n\
+            \                                                                                \n\
+            \Explanation: To translate to CSV you must provide a list of records.            \n\
+            \Other types can not be translated directly.                                     \n\
+            \                                                                                \n\
+            \Expected a record but instead got the following expression:                     \n\
+            \                                                                                \n\
+            \" <> insert e <>                                                               "\n\
+            \which has type:                                                                 \n\
+            \                                                                                \n\
+            \" <> insert (fromRight e (TypeCheck.typeOf e))
+
+    displayException BareNone =
+       Data.Text.unpack $
+            _ERROR <> ": ❰None❱ is not valid on its own                                      \n\
+            \                                                                                \n\
+            \Explanation: The conversion to JSON/YAML does not accept ❰None❱ in isolation as \n\
```

Comment:
I think this should say CSV instead of JSON/YAML

Reply:
Oops hahaha the dangers of copying and pasting. I'll change it right away